test/swar/BasicOperations.cpp

```diff
@@ -323,3 +337,76 @@ constexpr auto aBooleansWithTrue = booleans(SWAR<4, u32>{0x1});
 static_assert(aBooleansWithTrue);
 static_assert(!aBooleansWithTrue); // this is a pitfall, but lesser evil?
 static_assert(false == !bool(aBooleansWithTrue));
+
+TEST_CASE(
+    "fullAddition",
+    "[swar]"
+) {
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'1000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'8000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'F000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0101 (5) + 0x0101 (5) is 0x1010 (-x)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'5000), SWAR<4, u32>(0x0000'5000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'A000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0111 (7) is 0x1110 (0x1110->0x1101->0x0010) (e unsigned, 2 signed)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'7000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'e000).value() == fplusone.result.value());
+    }
+    {
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'a000), SWAR<4, u32>(0x0000'a000));
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
```

Comment:
I think I know why you are testing what you test here and not test other things: you're going over the cases for overflow.  Is that it? that would deserve to be explicitly stated in a comment at the heading; otherwise, the sample of tests seem arbitrary choices.